diff --git a/TangoCMakeModules/FindZeroMQ.cmake b/TangoCMakeModules/FindZeroMQ.cmake
index d5790ae..ef1a0d4 100644
--- a/TangoCMakeModules/FindZeroMQ.cmake
+++ b/TangoCMakeModules/FindZeroMQ.cmake
@@ -47,6 +47,10 @@ The following cache variables may also be set:
     ``ZeroMQ_static_LIBRARY``
     The path to the release ZeroMQ library, or the debug library
     if the release library is not found
+    ``ZeroMQ_RUNTIME_RELEASE``
+	The path to the release ZeroMQ dll, windows only.
+    ``ZeroMQ_RUNTIME_DEBUG``
+	The path to the debug ZeroMQ dll, windows only.
 
 #]=======================================================================]
 
@@ -58,10 +62,14 @@ endif()
 set(_ZeroMQ_PKG_INCLUDE_DIRS "")
 set(_ZeroMQ_PKG_LIBRARY_DIRS "")
 set(_ZeroMQ_PKG_CFLAGS_OTHER "")
+set(_ZeroMQ_PKG_PREFIX "")
 
 # Collect hints from pkg-config
 if (PKG_CONFIG_FOUND)
     pkg_search_module(_ZeroMQ_PKG libzmq QUIET)
+    if(NOT _ZeroMQ_PKG_LIBRARY_DIRS AND _ZeroMQ_PKG_LIBDIR)
+        set(_ZeroMQ_PKG_LIBRARY_DIRS ${_ZeroMQ_PKG_LIBDIR})
+    endif()
 endif()
 
 find_path(ZeroMQ_INCLUDE_DIR
@@ -70,7 +78,7 @@ find_path(ZeroMQ_INCLUDE_DIR
 )
 
 if(WIN32)
-    set(_zmq_versions "4_0_5" "4_3_4")
+    set(_zmq_versions "4_0_5" "4_3_4" "4_3_5")
     set(_zmq_vc_versions "" "-v141" "-v142")
 
     foreach(ver IN LISTS _zmq_versions)
@@ -79,6 +87,8 @@ if(WIN32)
             list(APPEND _zmq_lib_debug_names "libzmq${vc_ver}-mt-gd-${ver}.lib")
             list(APPEND _zmq_lib_static_release_names "libzmq${vc_ver}-mt-s-${ver}.lib")
             list(APPEND _zmq_lib_static_debug_names "libzmq${vc_ver}-mt-sgd-${ver}.lib")
+            list(APPEND _zmq_runtime_release_names "libzmq${vc_ver}-mt-${ver}.dll")
+            list(APPEND _zmq_runtime_debug_names "libzmq${vc_ver}-mt-gd-${ver}.dll")
         endforeach(vc_ver IN LISTS VC_VERSIONS)
     endforeach(ver IN ZMQ_VERSIONS)
 
@@ -94,33 +104,39 @@ endif(WIN32)
 find_library(ZeroMQ_LIBRARY_RELEASE
     NAMES ${_zmq_lib_release_names}
     PATHS "" ${_ZeroMQ_PKG_LIBRARY_DIRS}
+    PATH_SUFFIXES Release
 )
 
 find_library(ZeroMQ_LIBRARY_RELEASE
     NAMES ${_zmq_lib_static_release_names}
     PATHS "" ${_ZeroMQ_PKG_LIBRARY_DIRS}
+    PATH_SUFFIXES Release
 )
 
 find_library(ZeroMQ_LIBRARY_DEBUG
     NAMES ${_zmq_lib_debug_names}
     PATHS "" ${_ZeroMQ_PKG_LIBRARY_DIRS}
+    PATH_SUFFIXES Debug
 )
 
 find_library(ZeroMQ_LIBRARY_DEBUG
     NAMES ${_zmq_lib_static_debug_names}
     PATHS "" ${_ZeroMQ_PKG_LIBRARY_DIRS}
+    PATH_SUFFIXES Debug
 )
 
 find_library(ZeroMQ_static_LIBRARY_RELEASE
     NAMES ${_zmq_lib_static_release_names}
     NAMES_PER_DIR
     PATHS "" ${_ZeroMQ_PKG_LIBRARY_DIRS}
+    PATH_SUFFIXES Release
 )
 
 find_library(ZeroMQ_static_LIBRARY_DEBUG
     NAMES ${_zmq_lib_static_debug_names}
     NAMES_PER_DIR
     PATHS "" ${_ZeroMQ_PKG_LIBRARY_DIRS}
+    PATH_SUFFIXES Debug
 )
 
 unset(_zmq_lib_release_names)
@@ -128,6 +144,21 @@ unset(_zmq_lib_debug_names)
 unset(_zmq_lib_static_release_names)
 unset(_zmq_lib_static_debug_names)
 
+if(WIN32)
+    find_file(ZeroMQ_RUNTIME_DEBUG
+        NAMES ${_zmq_runtime_debug_names}
+        PATHS "${_ZeroMQ_PKG_PREFIX}"
+        PATH_SUFFIXES "bin/Debug" "bin"
+    )
+    find_file(ZeroMQ_RUNTIME_RELEASE
+        NAMES ${_zmq_runtime_release_names}
+        PATHS "${_ZeroMQ_PKG_PREFIX}"
+        PATH_SUFFIXES "bin/Release" "bin"
+    )
+    unset(_zmq_runtime_release_names)
+    unset(_zmq_runtime_debug_names)
+endif()
+
 include(SelectLibraryConfigurations)
 select_library_configurations(ZeroMQ)
 select_library_configurations(ZeroMQ_static)
